tests/conftest.py

```diff
@@ -834,38 +839,49 @@ def _process_cert(key: Crypto, cert: CertRequest, path_prefix: str):
     )
 
 
+def is_port_in_use(host: str, port: int) -> bool:
+    """Check if port is in use"""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex((host, port)) == 0
+
+
 def _make_libp2p_connection(
     data_dir: str,
-    port: int = 10234,
-    host: str = "127.0.0.1",
+    port: Optional[int] = None,
+    host: str = DEFAULT_HOST,
     relay: bool = True,
     delegate: bool = False,
     mailbox: bool = False,
     entry_peers: Optional[Sequence[MultiAddr]] = None,
-    delegate_port: int = 11234,
-    delegate_host: str = "127.0.0.1",
-    mailbox_port: int = 8888,
-    mailbox_host: str = "127.0.0.1",
+    delegate_port: Optional[int] = None,
+    delegate_host: str = DEFAULT_HOST,
+    mailbox_port: Optional[int] = None,
+    mailbox_host: str = DEFAULT_HOST,
     node_key_file: Optional[str] = None,
     agent_key: Optional[Crypto] = None,
     build_directory: Optional[str] = None,
     peer_registration_delay: str = "0.0",
 ) -> P2PLibp2pConnection:
-    if not os.path.isdir(data_dir) or not os.path.exists(data_dir):
+    """Get a libp2p connection."""
+
+    if not os.path.isdir(data_dir):
         raise ValueError("Data dir must be directory and exist!")
+
+    port = port or next(default_ports)
+    delegate_port = delegate_port or next(default_ports)
+    mailbox_port = mailbox_port or next(default_ports)
```

Comment:
port effectively always increments, as no test passes it. 

regarding delegate port and mailbox port: not optimal since we iterate always even if not used, but not a refactoring exercise so imo fine until refactor at a later date. Now one needs to explicitly pass matching ports in tests if they need to be (e.g. delegate connection port and node port for connection and client_connection). There was dependency on default values in functions here which is no longer the case

Reply:
I agree. In the particular case of ports, port 0 is not a valid port to use. If someone was to pass it though they could indeed get unexpected behaviour since it would be reassigned